sdcm/cluster_gce.py

```diff
@@ -152,16 +184,18 @@ def check_spot_termination(self):
     def restart(self):
         # When using local_ssd disks in GCE, there is no option to Stop and Start an instance.
         # So, for now we will keep restart the same as hard reboot.
-        self._instance_wait_safe(self._instance.reboot)
+        self._instance_wait_safe(self._gce_service.reset, instance=self._instance.name,
```

Comment:
shouln't we drop `_instance_wait_safe` in favor of `wait_for_extended_operation` (this one is used also for `.reset` method in GCE docs)
same for `hard_reboot`

Reply:
`_instance_wait_safe` is doing exponential backoff
`wait_for_extended_operation` is very specific and can working with only some api calls, and it won't every retry, it only for way for specific operation to finish.

so I'm not sure I'm ready to remove the those retries just yet, since I don't know the reason they were introduced for.